File: src/Samples/IPython/MinimalZeroInstall.py
import clr
clr.AddReferenceToFile("NanoByte.Common.dll", "ZeroInstall.Services.dll", "ZeroInstall.Store.dll")

import sys
from NanoByte.Common.Tasks import CliTaskHandler
from ZeroInstall.Services import ServiceLocator
from ZeroInstall.Store import FeedUri
from ZeroInstall.Store.Model import Requirements

services = ServiceLocator(CliTaskHandler())

selections = services.Solver.Solve(Requirements(sys.argv[1]))
missing = services.SelectionsManager.GetUncachedImplementations(selections)
services.Fetcher.Fetch(missing)
services.Executor.Start(selections)

File: src/Samples/VBasic/MinimalZeroInstall.vb
Imports NanoByte.Common.Tasks
Imports ZeroInstall.Services
Imports ZeroInstall.Store.Model

Module MinimalZeroInstall
    Sub Main(ByVal args As String())
        Dim services = New ServiceLocator(New CliTaskHandler())
        With services
            Dim selections = .Solver.Solve(New Requirements(args(0)))
            Dim missing = .SelectionsManager.GetUncachedImplementations(selections)
            .Fetcher.Fetch(missing)
            .Executor.Start(selections)
        End With
    End Sub
End Module